src/journal-remote/journal-header-util.c

```diff
@@ -0,0 +1,99 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "escape.h"
+#include "journal-header-util.h"
+#include "string-util.h"
+#include "strv.h"
+
+/* HTTP header name can contains:
+- Alphanumeric characters: a-z, A-Z, and 0-9
+- The following special characters: - and _ */
+#define VALID_HEADER_NAME_CHARS \
+        ALPHANUMERICAL "_-"
+
+#define VALID_HEADER_NAME_LENGTH 40
+
+#define VALID_HEADER_VALUE_CHARS \
+        ALPHANUMERICAL "_ :;.,\\/'\"?!(){}[]@<>=-+*#$&`|~^%"
+
+static bool header_name_is_valid(const char *e) {
+        if (!e)
+                return false;
+
+        if (strlen(e) == 0 || strlen(e) > VALID_HEADER_NAME_LENGTH)
+                return false;
+
+        return in_charset(e, VALID_HEADER_NAME_CHARS);
+}
+
+static bool header_value_is_valid(const char *e) {
+        if (!e)
+                return false;
+
+        return in_charset(e, VALID_HEADER_VALUE_CHARS);
```

Comment:
An empty value is allowed?

Reply:
https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.4
it's mentioned that Accept-Encoding header can be empty